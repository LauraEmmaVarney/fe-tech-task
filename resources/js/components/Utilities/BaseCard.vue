<template>
    <div class="card w-100 animated fadeInUp faster">
        <div class="card-header">
            <div class="row align-items-center">
                <slot name="header"></slot>
            </div>
        </div>
        <div 
            id="collapseOrder"
            class="" 
        >
            <div class="card-body p-0">
                <slot name="body"></slot>
            </div>
            <div class="card-footer">
                <slot name="footer"></slot>
            </div>
        </div>
    </div>
</template>

<script>
    export default {
        name: "BaseCard",
    }
</script>