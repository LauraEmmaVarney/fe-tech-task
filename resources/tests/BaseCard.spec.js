import { shallowMount } from '@vue/test-utils'
import BaseCard from '../js/components/Utilities/BaseCard.vue';

describe('BaseCard component tests', () => {
  const wrapper = shallowMount(BaseCard, {
    slots: {
      header: 'Your Details'
    }
  });

  it('mounts the component', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('checks the componets slots', () => {
    expect(wrapper.html()).toContain('Your Details');
  });

  it('check against jest snapshot test', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});