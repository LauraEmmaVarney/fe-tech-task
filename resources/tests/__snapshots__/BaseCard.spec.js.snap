// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseCard component tests check against jest snapshot test 1`] = `
<div
  class="card w-100 animated fadeInUp faster"
>
  <div
    class="card-header"
  >
    <div
      class="row align-items-center"
    >
      Your Details
    </div>
  </div>
   
  <div
    id="collapseOrder"
  >
    <div
      class="card-body p-0"
    />
     
    <div
      class="card-footer"
    />
  </div>
</div>
`;
